### PR TITLE
Re sync only disallowed to run when targeted sync is still actually running

### PR DESCRIPTION
**Pull request by wmathurin**

### CODE REVIEW

File: libs/SmartSync/src/com/salesforce/androidsdk/smartsync/manager/SyncManager.java [hunk 1]
```diff
@@ -289,6 +291,20 @@ private void updateSync(SyncState sync, SyncState.Status status, int progress, S
...
     		sync.save(smartStore);
+
+            switch (status) {
+
+                case NEW:
+                    break;
+                case RUNNING:
+                    runningSyncIds.add(sync.getId());
+                    break;
+                case DONE:
```
Comment (bhariharan) [inline libs/SmartSync/src/com/salesforce/androidsdk/smartsync/manager/SyncManager.java]: Meant to ask this earlier, but doesn't it have to be removed from the map when it gets to the ```DONE``` status?

Comment (wmathurin) [inline libs/SmartSync/src/com/salesforce/androidsdk/smartsync/manager/SyncManager.java]: For DONE and FAILED, it gets removed (it's falling through the case).

Comment (bhariharan) [inline libs/SmartSync/src/com/salesforce/androidsdk/smartsync/manager/SyncManager.java]: Oh right. I misread a nonexistent ```break;``` :-)

**Comment by bhariharan** [comment]
:+1:

**Pull request merged**